### PR TITLE
Player-controlled mobs are no longer stunned

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
This has been long overdue ever since that Snaxi round a year ago with the wendigo
As it turns out simple_animal mobs never have to worry about stuns because their rudimentary AI brute-forces it and moves mobs around itself, ignoring things such as whether the mob is stunned.
Clients on the other hand do not have such privilege, both the player-controlled movement and clicking are blocked if the mob is incapacitated in any way, meaning that in the end you COULD destroy all those admin-spawned player-controlled robo-hitlers at roundend Centcomm if you had a trusty taser with you.

:cl:
 * tweak: Player-controlled creatures (including russians) now ignore stuns as their AI-controlled counterparts do.

### CODE REVIEW

**Comment by ShiftyRail** [comment]
Player-controlled mobs also ignore the hard limit of a given number of tiles per `Life()` tick that their AI counterparts respect. Player simple mobs are very fast, their attack is only restrained by the universal click delay, combine that with their high damage output that ignores most armours and result in broken bones at the first hit.

If this PR is merged as is, this is the perfect recipe for never-ever busing player controlled simple mobs. Beyond that, there's the eventual issue of masks of morphing becoming suddenly much more powerful.
I really don't think this should be merged on its own.

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -1528,12 +1534,15 @@ Use this proc preferably at the end of an equipment loadout
 			canmove = 0
 			lying = (category.flags & LOCKED_SHOULD_LIE) ? TRUE : FALSE //A lying value that !=1 will break this
 
-
-	else if(isUnconscious() || knockdown || paralysis || resting || !can_stand)
+	else if(resting || !can_stand)
```
Comment (ShiftyRail) [inline code/modules/mob/mob.dm]: why do you put this in another `else` block if the executed code is exactly the same......

**Review by ShiftyRail — CHANGES_REQUESTED** [review]

File: code/modules/mob/mob.dm (hunk 1 above)
Comment (boy2mantwicethefam) [reply code/modules/mob/mob.dm]: isKnockedDown() will be overriden by a custom simple_animal check if there is a client present, meaning that if it wasn't for the separated check that does exactly the same resting would do nothing.

Comment (ShiftyRail) [reply code/modules/mob/mob.dm]:
```dm
else if(resting || !can_stand)
		stop_pulling()
		lying = 1
		canmove = 0
else if(isKnockedDown())
		stop_pulling()
		lying = 1
		canmove = 0
```
In what universe is this different from:
```dm
else if(resting || !can_stand || isKnockedDown())
		stop_pulling()
		lying = 1
		canmove = 0
```

**Comment by boy2mantwicethefam** [comment]
@ShiftyRail Most people don't even know that you can stun player-controlled mobs, you're claiming that the change is bad as if this extremely obscure oversight that almost nobody knows about is the only thing that keeps player-controlled simplemobs balanced even if it almost never comes into play, I don't want to get into pointless theorycrafting these days
Also something I've witnessed in a round while dead in-game (it happened after the PR was made): Juggernaut construct getting stunlocked to death by IAA briefcase, juggernaut in deadchat thought it was lag that did him in, it's that obscure.

File: code/modules/mob/mob.dm (hunk 1 above)
Comment (boy2mantwicethefam) [reply code/modules/mob/mob.dm]: Right

**Comment by ShiftyRail** [comment]
I agree that it's bad and obscure. Which is why the solution is a rework of how simple mob combat works. You can't just claim this one aspect is an oversight when nothing in it works.

**Review by ShiftyRail — APPROVED** [review]

**Comment by ShiftyRail** [comment]
Besides if you manage to land a melee hit on a jugger in melee they deserve to be stunned.

**Comment by help-maint** [comment]
By melee hit you mean flashbang from the edge of the screen

**Comment by boy2mantwicethefam** [comment]
It was an imperfect jugger aka no dash, some traitor made a TC trade for a soulstone belt and a construct shell

**Comment by Kammerjunk** [comment]
damn sounds like someone should be looking at flashbangs instead

**Comment by ShiftyRail** [comment]
Nevermind, I was wrong. That could be fixed, arguably, but it's part of a larger mob rework, again.

**Comment by d3athrow** [comment]
stun combat sucks anyways

**Comment by Kammerjunk** [comment]
not as much as proposed alternatives

**Comment by Kurfursten** [comment]
Reminds me of why you can never ever bus someone as a simplemob wizard

No cooldown on fireball

**Pull request merged**